### ISSUE TITLE
BUG: autoperf handling by "data access by category" infrastructure

### ISSUE THREAD

**Issue by tylerjereddy** [darshan-hpc/darshan#678]
Splitting off from gh-397. With code similar to this some of the autoperf records/data structure details produce "categories" that we probably don't want (see image below). I think @shanedsnyder has some idea for handling this.

```python
import darshan
from darshan.log_utils import get_log_path
from darshan.experimental.plots import data_access_by_filesystem

log_path = get_log_path("mpi-io-test.darshan")
report = darshan.DarshanReport(log_path)
fig = data_access_by_filesystem.plot_with_report(report=report,
                                                     num_cats=None)
fig.savefig("mpi-io-test.png", dpi=300)
```

![mpi-io-test darshan](https://user-images.githubusercontent.com/7903078/156698044-87bfcbde-f0ba-4f12-8d9c-8ee980ac8400.png)

**Comment by shanedsnyder** [comment]
For some background, the old darshan-job-summary tool had a "data transfer per filesystem" table that this new table is based off. It was intended to only show details on mount points used by applications (i.e., only relevant for Darshan modules that store "file" records, which does account for most of our current modules). We decided to generalize to account for things that aren't actually files, including the STDIO standard streams, but also could be used to characterize accesses to other things (e.g., the new Darshan DAOS modules could give access stats on different object storage pools, which are similar constructs to file system mounts, but index object IDs instead of file paths).

So, we ought to limit the records that are used to calculate rows in this table to modules that operate on files (or standard streams, I'll open an issue to describe that once I have a better idea of what others want). AutoPerf modules are capturing per-process details on network usage, so we should just ignore them entirely for this table. Actually, we should only need POSIX and STDIO module data for now (modules like MPI-IO, HDF5, etc., are layered on top of POSIX so counting them would lead to double counting). That would result in a table exclusively consisting of mount points, 1 or more rows for standard streams, and maybe one catch-all row "UNKNOWN" that could handle record names that don't match any mounts stored by Darshan (I'm thinking about cases in the past where we've anonymized standard stream names on accident, resulting in integer like record names -- I believe you have traditionally referred to these as "anonymized").

Once we have a good example of a module that's not doing file-based access (e.g., DAOS) merged into main, we can extend this to account for that type of data.

**Comment by tylerjereddy** [comment]
>  Actually, we should only need POSIX and STDIO module data for now

Looks like `plot_with_report()` is hardcoded to `POSIX` at the moment in gh-397.

**Comment by tylerjereddy** [comment]
For more concrete short term fixing purposes--in the small reproducer above, if you add:

```python
name_records = report.data["name_records"]
print("name_records:\n", name_records)
```

You get the origin of the weird rows:

```
name_records:
 {17895523780076722003: 'darshan-apxc-header', 14734109647742566553: '<STDIN>', 15920181672442173319: '<STDOUT>', 7238257241479193519: '<STDERR>', 4517659415186091864: 'APXC', 7713204788897956166: 'darshan-apmpi-header', 2250748219237254214: 'APMPI', 6138362456139396606: '/global/cscratch1/sd/ssnyder/tmp/mpi-io-test.tmp.dat'}
```

If I understand the comments above correctly, a "filter list" for dictionary/hash table keys we want to ignore might include most autoperf-related keys.

**Comment by shanedsnyder** [comment]
Right, the `name_records` dictionary is going to contain names for all modules contained in the log. I haven't grokked the code enough to understand how invasive of a change it might be, but I think instead what we want to do is start with all POSIX and STDIO module records (using their record IDs to find their corresponding names in `name_records`). That way we are only ever considering records (and record names) that are likely to be files or standard streams. I think it's going to be easier to restrict this sort of plot to specific module data (POSIX/STDIO), rather than creating an exclude list of record names from different modules ("darshan-apxc-header", etc.), if that makes sense?

**Comment by tylerjereddy** [comment]
Similar issue with `HEATMAP` module I think, using the tensorflow monitoring log from https://github.com/darshan-hpc/darshan-logs/pull/35


![image](https://user-images.githubusercontent.com/7903078/164754305-4b675199-c600-48f6-a8df-9db9bcc80142.png)

**Comment by shanedsnyder** [comment]
Closed via #722
